app/controllers/concerns/verify_setup_completed.rb

```diff
@@ -21,9 +21,9 @@ module VerifySetupCompleted
   ].freeze
 
   included do
-    # rubocop:disable Rails/LexicallyScopedActionFilter
-    before_action :verify_setup_completed, only: %i[index new edit show]
-    # rubocop:enable Rails/LexicallyScopedActionFilter
+    before_action :verify_setup_completed,
+                  only: %i[index new edit show],
+                  if: -> { current_user&.has_role?(:single_resource_admin) }
```

Comment:
Unsure if this is the right role to check for, maybe `admin` as well ?

I don't recall exactly what's the sequence of steps for a new forem

Reply:
The problem is that admins can't edit the config. So we'd show them a CTA for an action they can't perform.

Reply:
ah ok! thought admin/super admins could edit configuration without being single resource admins

Reply:
ah ok! thought admin/super admins could edit configuration without being single resource admins. makes sense to keep the two things separate

Reply:
So the way we currently have it is that you need to be a `super_admin` _and_ a `single_resource_admin, "Config"`, but I think that _might_ be overkill. We did it that way out of an abundance of caution, but realistically, `super_admin` should probably be enough. It would probably help to improve visualization of impending changes on an "are you sure" step, along with better safeguards on inputs that are particularly "dangerous".

..... But on this PR, I don't think this approach will work due to the caching of this page such that `current_user` would leak... But what if we moved this functionality into the `broadcast` area, which would DRY up the code anyway and allow more flexibility here, as that banner loads async.